src/fpm_meta.f90

```diff
@@ -1006,8 +1015,10 @@ subroutine init_mpi_from_wrappers(this,compiler,mpilib,fort_wrapper,c_wrapper,cx
 
     !> Add default run command, if present
     this%run_command = mpi_wrapper_query(mpilib,fort_wrapper,'runner',verbose,error)
-    if (allocated(error)) return
-    this%has_run_command = len_trim(this%run_command)>0
+    this%has_run_command = (len_trim(this%run_command)>0) .and. .not.allocated(error)
+
+    !> Do not trigger a fatal error here if run command is missing
+    if (allocated(error)) deallocate(error)
```

Comment:
We are ignoring the error here "if run command is missing", according to the comment. But could there be a case where the run command is not missing, but it couldn't be parsed, e.g. due to a typo? In that case, shouldn't we exit the program and inform the user instead of just setting `has_run_command` to `false`?

Reply:
I'm not in favor of the logic being dependent on the allocation status of the error here. If someone ever decides to change the error handling within `mpi_wrapper_query`, then the user will never receive the proper feedback. It should be implemented as stated in the comment (`Do not trigger a fatal error here if run command is missing`), irrespective of the allocation status of the error.

Reply:
@minhqdao I don't think this can easily be improved, but I like your idea. Can I ask you to fork this branch and add your changes to implement it?